data-explorer/kusto/query/toscalarfunction.md

```diff
@@ -31,10 +31,41 @@ If the result is a tabular, then the first column and first row will be taken fo
 > [!TIP]
 > You can use a [let statement](letstatement.md) for readability of the query when using `toscalar()`.
 
-**Notes**
+## Limitations
+
+`toscalar()` can be calculated a constant number of times during the query execution which means that the function can't be applied on row-level (for-each-row scenario).
+Usually, when this limitation is hit, the following error will be reported: `can't use '<column name>' as it is defined outside its row-context scope.`
+
+for example in this case, it will fail with the error `'toscalar': can't use 'x' as it is defined outside its row-context scope.` :
```

Comment:
I get the following error: 

 Semantic error: Unresolved reference binding: 'x'

Reply:
because there is an ongoing PR that should improve the error message :)
It was not merged yet as I need to generate a short link to this section and add it to the error message

Reply:
Ok great. Let me review the text first.